agents/ecloud/fence_ecloud.py

```diff
@@ -0,0 +1,201 @@
+            logging.error("timeout when performing {} action".format(action))
+            return 1
+        time.sleep(POWER_STATE_SLEEP_SECS)
+
+    logging.info("successfully performed action '{}' on '{}'".format(action, plug))
+    return 0
+
+
+def get_current_power_status(vpc, password, plug):
+    url = instance_data_url(vpc, plug)
+    hdrs = headers(password)
+
+    resp = requests.get(url, headers=hdrs)
+    if resp.status_code != 200:
+        logging.error("unexpected status code ('{}') from endpoint '{}': {}".format(
+            resp.status_code, url, resp.text
+        ))
+        return 1
+
+    instance = resp.json()['data']
+    if vpc:
+        return ON if instance['online'] else OFF
+    else:
+        if instance['power_status'] == "Online":
+            return ON
+        elif instance['power_status'] == "Offline":
+            return OFF
+        else:
+            # Could be 'Unknown' or other value
+            return None
+
+
+def monitor(password):
+    hdrs = headers(password)
+    resp = requests.get(API_MONITOR, headers=hdrs)
+    if resp.status_code != 200:
+        logging.error("eCloud API returned unexpected status code '{}'".format(resp.status_code))
+        return 1
+    if resp.text != "pong":
+        logging.error("eCloud API returned unexpected body from ping endpoint: '{}'".format(resp.text))
+        return 1
+    return 0
+
+
+def headers(password):
+    return {
+        "Authorization": password,
+        "User-Agent": "fence_ecloud"
+    }
+
+
+def itp(url, plug):
+    return url.replace(':ID', plug)
+
+
+def fence_url(vpc, action, plug):
+    if action == "on":
+        return itp(API_VPC_POWER_ON, plug) if vpc else itp(API_V1_POWER_ON, plug)
+    if action == "off":
+        return itp(API_VPC_POWER_OFF, plug) if vpc else itp(API_V1_POWER_OFF, plug)
+    
+    fail_usage("no available API configured for action '{}'".format(action))
+
+
+def instance_data_url(vpc, plug):
+    return itp(API_VPC_INSTANCE_DATA, plug) if vpc else itp(API_V1_INSTANCE_DATA, plug)
+
+
+def main():
+    device_opt = ["passwd", "port", "no_login"]
+
+    all_opt["passwd"]["help"] = "-p, --password=[apikey]        eCloud API Key"
+    all_opt["port"]["help"] = "-n, --plug=[instance]           Instance ID (VPC) or server ID (v1)"
+
+    atexit.register(atexit_handler)
+
+    options = check_input(device_opt, process_input(device_opt))
+    docs(options)
+
+    password = options['--password']
+    action = options['--action']
+
+    if action == 'monitor':
+        sys.exit(monitor(password))
+
+    plug = options['--plug']
+
+    if action in ["list", "list-status"]:
+        fail_usage("'{}' is not a valid action for this fence agent".format(options["--action"]))
+
+    if action == "validate-all":
+        sys.exit(0)
+
+    vpc = True
+    if not plug.startswith("i-"):
+        vpc = False
+
+    run_delay(options)
+
+    if action == 'status':
```

Comment:
You should use `fence_action()` like the other agents does: https://github.com/ClusterLabs/fence-agents/blob/main/agents/ilo/fence_ilo.py#L138